discord_bot/discord_bot.py

```diff
@@ -1,60 +1,88 @@
+import asyncio
 import json
 import requests
-import asyncio
 import websockets
 
+from .opcodes import Opcodes
+
+
 CONFIG_FILE_PATH = 'configs/config.json'
 DISCORD_GATEWAY_ENDPOINT = 'https://discordapp.com/api/v6/gateway'
 
 
 class DiscordBot:
-
+    """
+    Establishes a connection to the discord gateway and handles varied messages
+    """
     def __init__(self):
-
         # instance variables
         self.config = self.load_config()
         self.gateway_ws_url = self.get_gateway()
+        self.heartbeat_interval = None
+        self.last_seq = None
+        self.event_loop = asyncio.get_event_loop()
+        self.websocket = None
 
-        # actions below
-        asyncio.get_event_loop().run_until_complete(self.gateway_handler())
+        # actions
+        self.event_loop.run_until_complete(self.gateway_handler())
 
     @staticmethod
     def load_config():
+        """
+        Loads the configurations
+        :return: dict - the configurations
+        """
         with open(CONFIG_FILE_PATH) as f:
             return json.load(f)
 
     @staticmethod
     def get_gateway():
+        """
+        Caches a gateway value and retrieves a new URL
+        :return: gateway URL
+        """
         r = requests.get(DISCORD_GATEWAY_ENDPOINT)
```

Comment:
We should add this endpoint to the config file (and template)

Reply:
Done for both files
Added self to get_gateway